pkg/cmd/info.go

```diff
@@ -285,3 +310,119 @@ func retrieveConfigResourceNames(discoveryClient discovery.CachedDiscoveryInterf
 
 	return resources, nil
 }
+
+func (o *InfoOptions) gatherClusterOperatorNamespaceData(destDir, namespace string) error {
+	// ensure destination path exists
+	if err := os.MkdirAll(destDir, os.ModePerm); err != nil {
+		return err
+	}
+
+	ns, err := o.kubeClient.CoreV1().Namespaces().Get(namespace, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	ns.SetGroupVersionKind(corev1.SchemeGroupVersion.WithKind("Namespace"))
+
+	// write namespace.yaml file
+	filename := fmt.Sprintf("%s.yaml", namespace)
+	dest, err := os.OpenFile(path.Join(destDir, "/"+filename), os.O_RDWR|os.O_CREATE, 0755)
+	if err != nil {
+		return err
+	}
+	defer dest.Close()
+
+	if err := o.printer.PrintObj(ns, dest); err != nil {
+		return err
+	}
+
+	resourcesToStore := map[string]runtime.Object{}
+
+	// collect resource information for namespace
+	pods, err := o.kubeClient.CoreV1().Pods(namespace).List(metav1.ListOptions{})
```

Comment:
@deads2k I'm thinking I should probably use a dynamic client, right? That way I don't have to worry about setting GVK data for each list of resources

Reply:
> @deads2k I'm thinking I should probably use a dynamic client, right? That way I don't have to worry about setting GVK data for each list of resources

there is benefit to having real types that you can navigate. You're going to finding refs here.